### DISCUSSION TITLE
Can Smeared Cracking be Coupled with Production (Single Phase flow, Permeability Changes)

### DISCUSSION THREAD

**Discussion by miraboreasumr** [idaholab/moose#27894] [Q&A Modules: Solid mechanics]
I want to know, what features that MOOSE does't support while I am using smeared cracking (continuum damage mode) coupled with flow in porous media to simulate hydraulic fracturing propagation and then productions (flow in to the location of damage factor = 1).

I think I need to change the permeability at the grids with damage factor = 1, after this, I need to run another flow in porous media, or it can they be coupled? In-stitu stress needs to be considered.

Can you also see some convergence issue for coupling these two methods?

**Comment by GiudGiud** [comment]
@cpgr @rpodgorney

**Comment by rpodgorney** [comment]
Adding @lynnmunday , he has been doing smeared cracking runs in Falcon

**Comment by lynnmunday** [comment]
I used PorousFlowFullySaturated with coupling_type = HYDROMECHANICAL which will set up the kernels for coupling porous flow and mechanics i.e. PorousFlowEffectiveStressCoupling and PorousFlowVolumetricStrain and the mechanics kernel.  However, unlike the Physics/SolidMechanics action, this does not create a strain calculator material i.e. ComputeIncrementalSmallStrain, so you need to include that like I have below.


I have this auxvar for the smeared cracking strain that I use in the below to modify the permeability.
```
[AuxVariables]
  [./crack_max_strain1]
    order = CONSTANT
    family = MONOMIAL
  [../]
[]
[AuxKernels]
  [./crack_strain1]
    type = MaterialRealVectorValueAux
    property = crack_max_strain
    variable = crack_max_strain1
    component = 0
  [../]
[]
```

This is the auxkernel to modify the permeability.  I also tried to compute the aperture but it was 2 order magnitude too large.  That is why I'm using a made up relationship between permeability and smeared cracking strain instead of a^2/12.  The crack_perm function needs work, maybe it should look like the maybe_better_perm auxkernel but that gives too large of a crack.  A better measure of the element length normal to the crack face would give a better strain measure.
```
[AuxKernels]
[crack_perm]
  type = ParsedAux
  variable = crack_perm
  coupled_variables = 'permeability crack_max_strain1'
  expression = 'permeability *(1 + ${three_s_over_a0} * max(0, crack_max_strain1))'
  execute_on = NONLINEAR
[]
[maybe_better_perm]
  type = ParsedAux
  variable = crack_perm_not_used
  coupled_variables = 'permeability crack_max_strain1 elem_length_min'
  expression = 'a:=(crack_max_strain1*elem_length_min); permeability + (a)^2/12'
  execute_on = TIMESTEP_END
[]
[aperture]
  type = ParsedAux
  variable = aperture
  coupled_variables = 'crack_max_strain1 elem_length_min'
  expression = '(crack_max_strain1*elem_length_min)'
  execute_on = TIMESTEP_END
[]
[]
```

This is my porous flow and smeared cracking materials :
```
[Materials]
  [porosity]
    type = PorousFlowPorosityConst
    porosity = porosity 
  []
  [permeability]
    type = PorousFlowPermeabilityConstFromVar
    perm_xx = crack_perm
    perm_yy = crack_perm
    perm_zz = crack_perm
  []
  [elasticity_fracture]
    type = ComputeIsotropicElasticityTensor
    youngs_modulus = ${rock_young}
    poissons_ratio = ${rock_poisson}
  []
  [strain]
    type = ComputeIncrementalSmallStrain
    displacements = 'disp_x disp_y disp_z'
  []
  [elastic_stress]
    type = ComputeSmearedCrackingStress
    cracking_stress = 1.2e7
    softening_models = abrupt_softening
    max_stress_correction = 0.  #using zero lags damage by a timestep
    shear_retention_factor = 0.1
    cracked_elasticity_type = FULL
    output_properties = true
    perform_finite_strain_rotations=false
  []
  [abrupt_softening]
    type = AbruptSoftening
    residual_stress = 0.05
  []
  [undrained_density]
    type = GenericConstantMaterial
    prop_names = density
    prop_values = ${rock_gravitational_density}
  []
[]
```
To make this work, you need to define some kind of weakened field for the smeared cracking model to fracture along or a some distribution of initial damage.  Other wise you will get isotropic damage around the highest pressure point.  You can also use a spatially varying permeability to spread out damage

**Reply by lynnmunday** [reply]
Let me know if you get something working with the aperture.

**Reply by miraboreasumr** [reply]
> I used PorousFlowFullySaturated with coupling_type = HYDROMECHANICAL which will set up the kernels for coupling porous flow and mechanics i.e. PorousFlowEffectiveStressCoupling and PorousFlowVolumetricStrain and the mechanics kernel. However, unlike the Physics/SolidMechanics action, this does not create a strain calculator material i.e. ComputeIncrementalSmallStrain, so you need to include that like I have below.
> 
> I have this auxvar for the smeared cracking strain that I use in the below to modify the permeability.
> 
> ```
> [AuxVariables]
>   [./crack_max_strain1]
>     order = CONSTANT
>     family = MONOMIAL
>   [../]
> []
> [AuxKernels]
>   [./crack_strain1]
>     type = MaterialRealVectorValueAux
>     property = crack_max_strain
>     variable = crack_max_strain1
>     component = 0
>   [../]
> []
> ```
> 
> This is the auxkernel to modify the permeability. I also tried to compute the aperture but it was 2 order magnitude too large. That is why I'm using a made up relationship between permeability and smeared cracking strain instead of a^2/12. The crack_perm function needs work, maybe it should look like the maybe_better_perm auxkernel but that gives too large of a crack. A better measure of the element length normal to the crack face would give a better strain measure.
> 
> ```
> [AuxKernels]
> [crack_perm]
>   type = ParsedAux
>   variable = crack_perm
>   coupled_variables = 'permeability crack_max_strain1'
>   expression = 'permeability *(1 + ${three_s_over_a0} * max(0, crack_max_strain1))'
>   execute_on = NONLINEAR
> []
> [maybe_better_perm]
>   type = ParsedAux
>   variable = crack_perm_not_used
>   coupled_variables = 'permeability crack_max_strain1 elem_length_min'
>   expression = 'a:=(crack_max_strain1*elem_length_min); permeability + (a)^2/12'
>   execute_on = TIMESTEP_END
> []
> [aperture]
>   type = ParsedAux
>   variable = aperture
>   coupled_variables = 'crack_max_strain1 elem_length_min'
>   expression = '(crack_max_strain1*elem_length_min)'
>   execute_on = TIMESTEP_END
> []
> []
> ```
> 
> This is my porous flow and smeared cracking materials :
> 
> ```
> [Materials]
>   [porosity]
>     type = PorousFlowPorosityConst
>     porosity = porosity 
>   []
>   [permeability]
>     type = PorousFlowPermeabilityConstFromVar
>     perm_xx = crack_perm
>     perm_yy = crack_perm
>     perm_zz = crack_perm
>   []
>   [elasticity_fracture]
>     type = ComputeIsotropicElasticityTensor
>     youngs_modulus = ${rock_young}
>     poissons_ratio = ${rock_poisson}
>   []
>   [strain]
>     type = ComputeIncrementalSmallStrain
>     displacements = 'disp_x disp_y disp_z'
>   []
>   [elastic_stress]
>     type = ComputeSmearedCrackingStress
>     cracking_stress = 1.2e7
>     softening_models = abrupt_softening
>     max_stress_correction = 0.  #using zero lags damage by a timestep
>     shear_retention_factor = 0.1
>     cracked_elasticity_type = FULL
>     output_properties = true
>     perform_finite_strain_rotations=false
>   []
>   [abrupt_softening]
>     type = AbruptSoftening
>     residual_stress = 0.05
>   []
>   [undrained_density]
>     type = GenericConstantMaterial
>     prop_names = density
>     prop_values = ${rock_gravitational_density}
>   []
> []
> ```
> 
> To make this work, you need to define some kind of weakened field for the smeared cracking model to fracture along or a some distribution of initial damage. Other wise you will get isotropic damage around the highest pressure point. You can also use a spatially varying permeability to spread out damage

Hi, by any chance, is there an example that you can share with me so that I can understand the structure and make changes to make it work?

**Reply by lynnmunday** [reply]
I'll post it. I just need to clean it up first.

**Reply by miraboreasumr** [reply]
> I'll post it. I just need to clean it up first.

Hi Dr, do you have the chance to clean it?

**Comment by rpodgorney** [comment]
Low priority




Robert Podgorney, Ph.D.
Directorate Fellow, Energy and Environment
Idaho National Laboratory   |   1955 Fremont Ave.   |   Idaho Falls, ID   |   83415
***@***.******@***.***>   |   208-526-1524
_______________________________

[signature_4227069542]

Statement of Confidentiality
The contents of this e-mail message and any attachments are confidential and are intended solely for addressee. The information may also be legally privileged. This transmission is sent in trust, for the sole purpose of delivery to the intended recipient. If you have received this transmission in error, any use, reproduction or dissemination of this transmission is strictly prohibited. If you are not the intended recipient, please immediately notify the sender by reply e-mail or phone and delete this message and its attachments, if any.


From: Lynn Munday ***@***.***>
Date: Monday, June 17, 2024 at 13:42
To: idaholab/moose ***@***.***>
Cc: Robert K. Podgorney ***@***.***>, Mention ***@***.***>
Subject: [EXTERNAL] Re: [idaholab/moose] Can Smeared Cracking be Coupled with Production (Single Phase flow, Permeability Changes) (Discussion #27894)

I'll post it. I just need to clean it up first.

**Reply by lynnmunday** [reply]
I haven't yet. I"ll get this done probably next week.
